docs/tutorials/setup-opentelemetry.md

```diff
@@ -46,23 +48,68 @@ const sdk = new NodeSDK({
 sdk.start();
 ```
 
-In the `index.ts`, import this file **at the beginning**:
+Your probably won't need all the instrumentation inside `getNodeAutoInstrumentations()` so make sure to
+check the [documentation](https://www.npmjs.com/package/@opentelemetry/auto-instrumentations-node) and tweak it properly.
 
-```typescript
-import './instrumentation'; // Setup the OpenTelemetry instrumentation
+It's important to setup the NodeSDK and the automatic instrumentation **before** importing any library.
 
-// other imports and backend init...
-```
+This is why we will use the nodejs [`--require`](https://nodejs.org/api/cli.html#-r---require-module)
+flag when we start up the application.
 
-It's important to setup the NodeSDK and the automatic instrumentation **before** importing any library.
+In your `Dockerfile` add the `--require` flag which points to the `instrumentation.ts` file
+
+```Dockerfile
+FROM node:18-bookworm-slim
+...
+# More functionality goes here
+...
+WORKDIR /app
+RUN chown node:node /app
+USER node
+
+ENV NODE_ENV production
+
+COPY --chown=node:node .yarn ./.yarn
+COPY --chown=node:node .yarnrc.yml ./
+
+# We need the instrumentation file inside the Docker image so we can use it with --require
+// highlight-add-next-line
+COPY --chown=node:node packages/backend/instrumentation.ts ./
+
+COPY --chown=node:node yarn.lock package.json packages/backend/dist/skeleton.tar.gz ./
+RUN tar xzf skeleton.tar.gz && rm skeleton.tar.gz
+
+RUN --mount=type=cache,target=/home/node/.yarn/berry/cache,sharing=locked,uid=1000,gid=1000 \
+    yarn workspaces focus --all --production
+
+COPY --chown=node:node packages/backend/dist/bundle.tar.gz app-config*.yaml ./
+RUN tar xzf bundle.tar.gz && rm bundle.tar.gz
+
+// highlight-remove-next-line
+CMD ["node", "packages/backend", "--config", "app-config.yaml"]
+// highlight-add-next-line
+CMD ["node", "--require", "./instrumentation.ts", "packages/backend", "--config", "app-config.yaml"]
```

Comment:
I'd say remove everything above except this removal and addition. It's going to quickly diverge from reality and how the actual Dockerfile looks otherwise, over time.